### PR TITLE
Add download_folder method

### PR DESCRIPTION
**Pull request by RTLcoil**
Add a new method in the Upload API - `download_folder`.

This method creates and returns a URL that when invoked creates an archive of a folder.

### CODE REVIEW

File: lib/cloudinary/utils.rb [hunk 1]
```diff
@@ -969,7 +969,8 @@ def self.archive_params(options = {})
       :expires_at=>options[:expires_at],
       :transformations => build_eager(options[:transformations]),
       :skip_transformation_name=>Cloudinary::Utils.as_safe_bool(options[:skip_transformation_name]),
-      :allow_missing=>Cloudinary::Utils.as_safe_bool(options[:allow_missing])
+      :allow_missing=>Cloudinary::Utils.as_safe_bool(options[:allow_missing]),
+      :folder_path=>options[:folder_path]
```
Comment (const-cloudinary) [inline lib/cloudinary/utils.rb:973]: I don't quite understand where this key is used.

File: lib/cloudinary/utils.rb [hunk 2]
```diff
@@ -996,7 +997,22 @@ def self.generate_auth_token(options)
...
+  #
+  # @param [Object] folder_path Full path (from the root) of the folder to download.
+  # @param [Hash] options       Additional options.
+  #
+  # @return [String]
+  def self.download_folder(folder_path, options = {})
+    options[:resource_type] = options[:resource_type] || "all"
+    options[:prefixes] = folder_path
+
+    cloudinary_params = Cloudinary::Utils.sign_request(Cloudinary::Utils.archive_params(options.merge(:mode => "download")), options)
```
Comment (const-cloudinary) [inline lib/cloudinary/utils.rb:1011]: You can call `download_archive_url` directly, in a similar way it is used in `download_zip_url`

File: lib/cloudinary/utils.rb [hunk 3]
```diff
@@ -996,7 +997,22 @@ def self.generate_auth_token(options)
...
   end
-  
+
+  # Creates and returns a URL that when invoked creates an archive of a folder.
+  #
+  # @param [Object] folder_path Full path (from the root) of the folder to download.
+  # @param [Hash] options       Additional options.
+  #
+  # @return [String]
+  def self.download_folder(folder_path, options = {})
```
Comment (const-cloudinary) [inline lib/cloudinary/utils.rb:1007]: Please move this function closer to `download_archive_url`, it's just an alias with prepopulated `prefix` and `resource_type`.

File: spec/utils_spec.rb [hunk 4]
```diff
@@ -1103,4 +1103,37 @@
 
   end
   end
+
+  describe "download_folder" do
```
Comment (const-cloudinary) [inline spec/utils_spec.rb:1107]: Please move these tests to `archive_spec.rb`

File: lib/cloudinary/utils.rb (hunk 1 above)
Comment (RTLcoil) [reply lib/cloudinary/utils.rb:973]: @const-cloudinary 
Correct me if I'm wrong, but I believe that according to the reference implementation [this param](https://github.com/cloudinary/cloudinary_npm/blob/7aba2a9f6f8bafe1cb627b70e5f225cdf0b43e25/lib/utils/index.js#L1213) is one of the params used [when creating the params object and signing the request](https://github.com/cloudinary/cloudinary_npm/blob/7aba2a9f6f8bafe1cb627b70e5f225cdf0b43e25/lib/utils/index.js#L1033). I think this change is required, no?

Comment (const-cloudinary) [reply lib/cloudinary/utils.rb:973]: @RTLcoil That is a bug, there is no such parameter `folder_path` in the API, it is never passed to the actual API call. You can open an issue on that.

**Pull request merged**